#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.22990370747049949 0.9696956345388842 0.08267201248034094 0.08305616505938494
  position 1.1991748735771937 0.29516118510072104 4.896727626579817
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 20 20
}
Robot {
  translation 0 0.1 0
  children [
    DEF GR_BODY Group {
      children [
        DEF ROBOT_BODY Shape {
          appearance PBRAppearance {
            baseColor 0.9294117647058824 0.8313725490196079 0
          }
          geometry DEF G_BODY Box {
            size 0.25 0.07 0.1
          }
        }
        Transform {
          translation 0.11 -0.05 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.12549019607843137 0.2901960784313726 0.5294117647058824
                roughness 1
                metalness 0
              }
              geometry Sphere {
                radius 0.015
              }
            }
          ]
        }
      ]
    }
    DEF WHEEL1_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position 1
        axis 0 0 1
        anchor -0.05 0 0.06
      }
      device [
        PositionSensor {
          name "encoder1"
        }
        RotationalMotor {
          name "motor_right"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation -0.05 0 0.06
        rotation 0.7913386706458939 0.4323102860835271 0.43230883045199103 1.8027213601447765
        children [
          DEF SH_WHEEL Shape {
            appearance DEF A_WHEEL PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry DEF G_WHEEL Cylinder {
              height 0.02
              radius 0.075
            }
          }
        ]
        boundingObject USE G_WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.05 0 -0.06
      }
      device [
        RotationalMotor {
          name "motor_left"
        }
      ]
      endPoint Solid {
        translation -0.05 0 -0.06
        rotation -1 0 0 1.5708003061004252
        children [
          Shape {
            appearance USE A_WHEEL
            geometry USE G_WHEEL
          }
        ]
        name "solid(1)"
        boundingObject USE G_WHEEL
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation 0.125 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.807843137254902 0.3607843137254902 0
          }
          geometry Box {
            size 0.02 0.01 0.01
          }
        }
      ]
      name "distance_sensor"
      locked TRUE
      radarCrossSection 1.5708
      lookupTable [
        0 0 0
        0.02 0 0
        11 1659 0
      ]
      resolution 1
    }
  ]
  boundingObject USE GR_BODY
  physics Physics {
  }
  controller "U2_exam_Chacon_Ivan"
}
Solid {
  translation 2 0.225 0
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.788235294117647 0 0.7137254901960784
        roughness 1
        metalness 0
      }
      geometry DEF G_BOX Box {
        size 0.5 0.5 0.5
      }
    }
  ]
  boundingObject USE G_BOX
  physics Physics {
  }
}
